tests/server/test_hot_reloading.py

```diff
@@ -166,82 +164,48 @@ def test_file_monitoring(delete_mock, post_mock, tmp_path: Path):
             yaml.safe_load(content)
 
 
-@pytest.fixture
-def server_port():
-    p = _find_free_port()
-    yield p
-
-
-@pytest.fixture
-def config_path(tmp_path: Path):
+@mock.patch("uvicorn.run")
+@mock.patch("deepsparse.server.server.start_config_watcher")
+def test_hot_reload_config_with_start_server(
+    watcher_patch: mock.Mock, run_patch: mock.Mock, tmp_path: Path
+):
     cfg = ServerConfig(endpoints=[], num_cores=1, num_workers=1, loggers=None)
     cfg_path = str(tmp_path / "cfg.yaml")
     with open(cfg_path, "w") as fp:
         yaml.safe_dump(cfg.dict(), fp)
-    yield cfg_path
-
-
-@pytest.fixture
-def server_process(config_path, server_port):
-    proc = multiprocessing.Process(
-        target=start_server,
-        kwargs=dict(
-            config_path=config_path,
-            host="127.0.0.1",
-            port=server_port,
-            hot_reload_config=True,
-        ),
-    )
-    proc.start()
-    yield proc
-    proc.kill()
-    proc.join()
-
-
-def test_hot_reload_config_with_start_server(server_process, server_port, config_path):
-    # wait max 5s for server to spin up
-    assert wait_for_server(f"http://127.0.0.1:{server_port}", retries=50, interval=0.1)
-
-    resp = requests.get(f"http://127.0.0.1:{server_port}/")
-    assert resp.status_code == 200
-
-    # endpoint doesn't exist yet
-    resp = requests.post(f"http://127.0.0.1:{server_port}/predict1")
-    assert resp.status_code == 404
-
-    cfg = ServerConfig(endpoints=[], num_cores=1, num_workers=1, loggers=None)
-    cfg.endpoints.append(
-        EndpointConfig(
-            route="/predict1",
-            task="qa",
-            model="default",
-            kwargs=dict(engine_type="onnxruntime"),
-        )
-    )
-    with open(config_path, "w") as fp:
-        yaml.safe_dump(cfg.dict(), fp)
-
-    # wait for endpoint to be added to server
-    for _ in range(50):
-        time.sleep(0.1)
-        resp = requests.post(f"http://127.0.0.1:{server_port}/predict1")
-        if resp.status_code != 404:
-            break
-
-    # the endpoint should exist now
-    resp = requests.post(
-        f"http://127.0.0.1:{server_port}/predict1",
-        json={"question": "who am i", "context": "i am bob"},
-    )
-    assert resp.status_code == 200
-
-    cfg.endpoints.clear()
-    with open(config_path, "w") as fp:
-        yaml.safe_dump(cfg.dict(), fp)
-
-    # wait for endpoint to be removed from the server
-    time.sleep(1.0)
 
-    # the endpoint should not exist anymore
-    resp = requests.post(f"http://127.0.0.1:{server_port}/predict1")
-    assert resp.status_code == 404
+    start_server(cfg_path, hot_reload_config=False)
+    run_patch.assert_called_once()
+    watcher_patch.assert_not_called()
+
+    run_patch.reset_mock()
+    watcher_patch.reset_mock()
+    start_server(cfg_path, hot_reload_config=True)
+    run_patch.assert_called_once()
+    watcher_patch.assert_called_once()
```

Comment:
let's add in some inline comments for what's being tested

Reply:
Good point, added comments and improved skimmability of them